go/vt/vtgate/engine/set.go

```diff
@@ -445,8 +445,6 @@ func (svss *SysVarSetAware) Execute(ctx context.Context, vcursor VCursor, env *e
 	switch svss.Name {
 	case sysvars.Autocommit.Name:
 		err = svss.setBoolSysVar(ctx, env, vcursor.Session().SetAutocommit)
-	case sysvars.ForeignKeyChecks.Name:
-		err = svss.setBoolSysVar(ctx, env, vcursor.Session().SetSessionForeignKeyChecks)
```

Comment:
This was the only place `SetSessionForeignKeyChecks` was being used in. We can remove it from the interface `SessionActions`